### ISSUE TITLE
Measure: wrong measurements in nested parts with linked objects

### ISSUE THREAD

**Issue by johndoe2323** [FreeCAD/FreeCAD#16058]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

1. Create part
2. Within part insert standard cube in Part workbench
3. Create Link (Std_LinkMake) to first cube within the same part
4. Move the linked cube e.g. y-position to 13mm
5. Insert measurement (Std_Measure) between two edges of the original cube and the linked cube
6. Move the part e.g. y-position to 10mm

Issue: the measurement is wrong and not positioned correctly (e.g. should be 3mm in the example but gives 7mm).

![image](https://github.com/user-attachments/assets/b51eeb2a-3d13-49c0-abd7-a34c01ec6700)

### Full version info

```shell
OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38506 (Git)
Build type: Release
Branch: main
Hash: ebf02483af225b502fccfa7111499dfde33999fd
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: German/Austria (de_AT)
Installed mods: 
  * Assembly4 0.50.4
  * OpenDark 2023.10.30
  * ProDarkThemePreferencePack 1.0.0
  * Woodworking 0.21.30486
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hlorus** [comment]
I cannot reproduce this, also this seems like a duplicate of #13718 and should be resolved since the corresponding PR got merged.

**Comment by luzpaz** [comment]
@hlorus fix was made by you in #15874 which is aa98951b5c75839eacb30d5455986e733df26d7d 

Running a quick conversion on commit -> revision number 
```shell
$ git rev-list --count aa98951b5c75839eacb30d5455986e733df26d7d
38490
```

That fix was in v0.22.**38490**.  
@johndoe2323 ticket was  0.22.0dev.**38506**. So this fix should be in their revision.   

@johndoe2323 can you download latest dev version and retest again?

**Comment by johndoe2323** [comment]
@luzpaz: unfortunately same behaviour in 38553, issue should be reopened

![image](https://github.com/user-attachments/assets/19f4594c-6e4f-42cb-ba7c-95ebce783bf7)

**Comment by maxwxyz** [comment]
@hlorus could you check it again please?

**Comment by luzpaz** [comment]
@johndoe2323 can you add your full [About](https://wiki.freecad.org/About) info please? TIA

**Comment by johndoe2323** [comment]
@luzpaz 

OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38553 (Git)
Build type: Release
Branch: main
Hash: 59c1ccec3e6b70f56eeee8f94d361019b84bd850
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: German/Austria (de_AT)
Installed mods: 
  * Assembly4 0.50.4
  * OpenDark 2023.10.30
  * ProDarkThemePreferencePack 1.0.0
  * Woodworking 0.21.30486

**Comment by johndoe2323** [comment]
[16058 sample for issue std_measure nested part.zip](https://github.com/user-attachments/files/16824781/16058.sample.for.issue.std_measure.nested.part.zip)

**Comment by NomAnor** [comment]
I can confirm, fresh build of master:
```
OS: Arch Linux (XFCE/xfce)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38513 (Git)
Build type: Release
Branch: main
Hash: 7c797b5a10c19ae55e317c4ff9a02823a4b4dc0b
Python 3.12.5, Qt 6.7.2, Coin 4.0.2, Vtk 9.3.1, OCC 7.7.2
Locale: German/Germany (de_DE)
Installed mods: 
  * Manipulator 1.5.7
  * ThreadProfile 1.84.0
  * DynamicData 2.62.0
  * Assembly3.backup1715897154.0970206 0.11.3 (Disabled)
  * lattice2 1.0.0
  * 3D_Printing_Tools
  * freecad.gears 1.2.0
  * fasteners 0.5.27
```

![measurement](https://github.com/user-attachments/assets/41280081-54ea-4c9f-a0a3-36322d49a200)

**Comment by johndoe2323** [comment]
Can confirm, bug is resolved in 1.1.0dev.38794! Many thanks @hlorus, you guys are awesome!

OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.38794 (Git)
Build type: Release
Branch: main
Hash: a47800d9ccd7d7a7cdb1779d56de03d3c64cd76e
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: German/Austria (de_AT)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * Assembly4 0.50.4
  * OpenDark 2023.10.30
  * ProDarkThemePreferencePack 1.0.0
  * Woodworking 0.21.30486
